waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -647,10 +647,12 @@
      :watch-state @watch-state})
 
   (validate-service [_ service-id]
-    (let [run-as-user (-> service-id
-                          service-id->service-description-fn
-                          service-description->namespace)]
-      (authz/check-user authorizer run-as-user service-id))))
+    (let [{:strs [ports] :as service-description} (service-id->service-description-fn service-id)]
+      (let [run-as-user (service-description->namespace service-description)]
+        (authz/check-user authorizer run-as-user service-id))
+      (when (and (scheduler/http2-service? service-description) (ports < 2))
```

Comment:
Kubernetes requires an HTTP/2 service to listen on `PORT1` for health checks with HTTP/1 requests.